### ISSUE TITLE
preprocessing VoxCele2 is not working

### ISSUE THREAD

**Issue by amintavakol** [CorentinJ/Real-Time-Voice-Cloning#488]
While running encoder_preprocess on voxceleb2 dataset, I'm getting the following warning and nothing else happens. Not sure why?

```
raw: Preprocessing data for 5994 speakers.
raw:   0%|                                                                                           | 0/5994 [00:00<?, ?speakers/s]
/home/amin/.local/lib/python3.6/site-packages/librosa/core/audio.py:161: UserWarning: PySoundFile failed. Trying audioread instead.
  warnings.warn('PySoundFile failed. Trying audioread instead.')
/home/amin/.local/lib/python3.6/site-packages/librosa/core/audio.py:161: UserWarning: PySoundFile failed. Trying audioread instead.
  warnings.warn('PySoundFile failed. Trying audioread instead.')
```

**Comment by blue-fish** [comment]
Could it be related to this? https://github.com/CorentinJ/Real-Time-Voice-Cloning/issues/76#issuecomment-529013562

Also you might want to follow #458 since we're also running into issues with preprocessing and training. If you modify code to fix a problem would you please contribute it back as a pull request?

**Comment by mbdash** [comment]
it needs to be converted to wav

**Comment by mbdash** [comment]
I have mesed around with LibriSpeeech, VoxCeleb 1&2, CommonVoice and VCTK in https://github.com/CorentinJ/Real-Time-Voice-Cloning/issues/458

I am a bit tired so I am not sure if VoxCeleb was wav, ensure it is all converted to wav.
Also, once I succeed in training, a new encoder, I will share everything i can.

**Comment by amintavakol** [comment]
For those who are trying to preprocess VoxCeleb2:
once you download the dataset, the audio files are in ".m4a" format. You guys need to reformat the audio files into ".wav".
Just put the following code snippet (`convert.sh`) in the root directory of the data (e.g. `<path-to-VoxCeleb2>/raw/dev/aac`)

`
open_sem(){
    mkfifo pipe-$$
    exec 3<>pipe-$$
    rm pipe-$$
    local i=$1
    for((;i>0;i--)); do
        printf %s 000 >&3
    done
}
run_with_lock(){
    local x
    read -u 3 -n 3 x && ((0==x)) || exit $x
    (
     ( "$@"; )
    printf '%.3d' $? >&3
    )&
}

N=32 # number of vCPU
open_sem $N
for f in $(find . -name "*.m4a"); do
    run_with_lock ffmpeg -loglevel panic -i "$f" -ar 16000 "${f%.*}.wav"
done
                       `

then run 
`./convert.sh`

Also, make sure you have ffmpeg installed on you machine.
You also need to modify the function  `preprocess_voxceleb2` in `encode/preprocess.py`, and change the extension to ".wav".

I'm training a new encoder with more datasets other than Libri/Vox1,2 and will update everyone in a few days.

**Comment by blue-fish** [comment]
Presumed resolved based on #497

**Comment by blue-fish** [comment]
Please share updates about your encoder model in #458, would be interested to see how it is working. Did you modify any hparams?
